common/lib/client/channel.ts

```diff
@@ -0,0 +1,178 @@
+		const cipher = Crypto.getCipher(channelOptions.cipher);
+		channelOptions.cipher = cipher.cipherParams;
+		channelOptions.channelCipher = cipher.cipher;
+	} else if('cipher' in channelOptions) {
+		/* Don't deactivate an existing cipher unless options
+			* has a 'cipher' key that's falsey */
+		channelOptions.cipher = null;
+		channelOptions.channelCipher = null;
+	}
+	return channelOptions;
+}
+
+class Channel extends EventEmitter {
+	rest: Rest | Realtime;
+	name: string;
+	basePath: string;
+	presence: Presence;
+	channelOptions: ChannelOptions;
+
+	constructor(rest: Rest | Realtime, name: string, channelOptions?: ChannelOptions) {
+		super();
+		Logger.logAction(Logger.LOG_MINOR, 'Channel()', 'started; name = ' + name);
+		this.rest = rest;
+		this.name = name;
+		this.basePath = '/channels/' + encodeURIComponent(name);
+		this.presence = new Presence(this);
+		this.channelOptions = {};
+		this.channelOptions = normaliseChannelOptions(channelOptions);
+	}
+
+	setOptions(options: ChannelOptions): void {
+		this.channelOptions = normaliseChannelOptions(options);
+	}
+
+	history(params: RestHistoryParams | null, callback: PaginatedResultCallback<Message>): Promise<PaginatedResult<Message>> | void {
+		Logger.logAction(Logger.LOG_MICRO, 'Channel.history()', 'channel = ' + this.name);
+		/* params and callback are optional; see if params contains the callback */
+		if(callback === undefined) {
+			if(typeof(params) == 'function') {
+				callback = params;
+				params = null;
+			} else {
+				if(this.rest.options.promises) {
+					return Utils.promisify(this, 'history', arguments);
+				}
+				callback = noop;
+			}
+		}
+
+		this._history(params, callback);
+	}
+
+	_history(params: RestHistoryParams | null, callback: PaginatedResultCallback<Message>): void {
+		const rest = this.rest,
+			format = rest.options.useBinaryProtocol ? Utils.Format.msgpack : Utils.Format.json,
+			envelope = Http.supportsLinkHeaders ? undefined : format,
+			headers = Utils.defaultGetHeaders(format);
+
+		if(rest.options.headers)
+			Utils.mixin(headers, rest.options.headers);
+
+		const options = this.channelOptions;
+		(new PaginatedResource(rest, this.basePath + '/messages', headers, envelope, function(body: any, headers: Record<string, string>, unpacked?: boolean) {
+			return Message.fromResponseBody(body, options, unpacked ? undefined : format);
+		})).get(params as Record<string, unknown>, callback);
+	}
+
+	publish(): void | Promise<void> {
+		const argCount = arguments.length,
+			first = arguments[0],
+			second = arguments[1];
+		let callback = arguments[argCount - 1];
+		let messages: Array<Message>;
+		let params: any;
+
+		if(typeof(callback) !== 'function') {
+			if(this.rest.options.promises) {
+				return Utils.promisify(this, 'publish', arguments);
+			}
+			callback = noop;
+		}
+
+		if(typeof first === 'string' || first === null) {
+			/* (name, data, ...) */
```

Comment:
This comment may not be needed 🤔

Reply:
I think it's helpful, it's just saying that if we reach this branch the first argument is the event name and the second argument is the data to publish

Reply:
Resolving this conversation on @ikbalkaya's behalf.